pkg/kubelet/dockertools/manager.go

```diff
@@ -1170,6 +1171,9 @@ func (dm *DockerManager) killContainer(containerID types.UID) error {
 					break
 				}
 			}
+			if pod.Spec.TerminationGracePeriodSeconds != nil {
+				gracePeriod = uint(*pod.Spec.TerminationGracePeriodSeconds)
```

Comment:
> On May 21, 2015, at 7:25 PM, Brian Grant <notifications@github.com> wrote:
> 
> In pkg/kubelet/dockertools/manager.go:
> 
> > @@ -1170,6 +1171,9 @@ func (dm *DockerManager) killContainer(containerID types.UID) error {
> >  					break
> >  				}
> >  			}
> > +			if pod.Spec.TerminationGracePeriodSeconds != nil {
> > +				gracePeriod = uint(*pod.Spec.TerminationGracePeriodSeconds)
> Why aren't we using DeletionTimestamp here instead? Or do we need to convert that to a duration?
> 
DeletionTimestamp is important for history (mostly) and clients.  I think spec being changed fits with other uses, so I'd rather not have two durations on the object.
> I expected:
> 1. User sets terminationGracePeriodSeconds or it gets set to 30s by default
> 2. On deletion, deletionTimestamp is set, to Now+terminationGracePeriodSeconds by default by controllers
> 3. Upon noticing deletionTimestamp is set, prestop hook is executed
> 4. After prestop hook completes, stop is called with time deletionTimestamp-Now
> 
The timestamp has to be measured from the local clock, so it's similar to Derek's changes that the local time matters more.
> Eventually we want to "right align" notifications, but the above seems like a reasonable starting point.
> 
> —
> Reply to this email directly or view it on GitHub.
>